### PR TITLE
fix(#1107): drop support for *** for thematic-break

### LINKED ISSUE #1107: Small parser problem : bold + italics don't work if nothing before in a block
**Problem**

Bold + Italics won't work if there is nothing before in a block
Couldn't find any other combo that doesn't work except bold + italics + something inside.

* `***Test***` doesn't work
* `***^^Test^^***` won't work

When hitting `Enter` the block will split after the three `*` and make a new line.

However, all other combinations, including when there is text before or when the nesting of rendering commands is in a different order : 

* `^^***Test***^^` works
* `Hello ***Test***` works

I've also noticed two strange things : 
* `* Test` will be rendered as `* Test`
* `** Test` will be rendered as `Test` (double star has disappeared)
* `*** Test` will produce the same behaviour by splitting the block in two lines

**Athens Version**
1.0.0-BETA.78 - Intel OSX 10.11.6 - El Capitan

### PR DESCRIPTION
**Pull request by zkry**
This PR addresses issue #1107. 

The root cause was that the block parser was interpreting a `***` to be a thematic block which in turn would cause something like `***bold and emphasis***` to be `[:block [:thematic-block "***"] [:paragraph-text "bold and emphasis***"]]`.  I added a negative lookahead in the regex of thematic block to prevent it from matching if there is a `***` which could be interpreted as the end of an emphasis + strong-emphasis.

### CODE REVIEW

**Comment by tangjeff0** [comment]
@neotyk please review

**Comment by neotyk** [comment]
@zkry Thank you for PR.

Maybe we could just drop support for `***` as thematic break?

File: src/cljc/athens/parser/impl.cljc [hunk 1]
```diff
@@ -24,7 +24,7 @@ block = (thematic-break /
          fenced-code-block /
          block-quote /
          paragraph-text)*
-thematic-break = #'[*_-]{3}'
+thematic-break = #'(\\*{3}(?!\\w[^\n]*(?<!\\s)\\*{3}(?!\\w))|[_-]{3})'
```
Comment (neotyk) [inline src/cljc/athens/parser/impl.cljc:27]: That's a lot of regex.
I think we could drop support of `***` as thematic break.

**Review by neotyk — CHANGES_REQUESTED** [review]

File: src/cljc/athens/parser/impl.cljc (hunk 1 above)
Comment (zkry) [reply src/cljc/athens/parser/impl.cljc:27]: I agree, it makes more sense to do this. I can update the PR for this.

**Comment by neotyk** [comment]
LGTM

**Review by tangjeff0 — CHANGES_REQUESTED** [review]
Failing test

**Review by tangjeff0 — APPROVED** [review]
Awesome first PR @zkry :muscle: !!!

**Pull request merged**